apps/site/lib/site_web/controllers/stop_controller.ex

```diff
@@ -177,6 +179,44 @@ defmodule SiteWeb.StopController do
     json(conn, Repo.get(stop_id))
   end
 
+  @spec grouped_route_patterns(Conn.t(), map) :: Conn.t()
+  def grouped_route_patterns(conn, %{"id" => stop_id}) do
+    json(conn, route_patterns_by_route_and_headsign(stop_id))
+  end
+
+  @type by_route_and_headsign :: %{Route.id_t() => %{String.t() => [RoutePattern.t()]}}
+  @spec route_patterns_by_route_and_headsign(Stop.id_t()) :: by_route_and_headsign()
+  defp route_patterns_by_route_and_headsign(stop_id) do
+    stop_id
+    |> RoutePatterns.Repo.by_stop_id()
+    |> Enum.reject(&ends_at?(&1, stop_id))
```

Comment:
I believe there was a check for "is at last stop" on the JS side. Has that been removed since it's happening here?

Reply:
@i0sea Yep, it's been removed!